datalab_cohorts/__init__.py

```diff
@@ -1637,19 +1586,48 @@ def with_death_recorded_in_cpns(
         between=None,
         # Set return type
         returning="binary_flag",
-        # If we're returning a date, how granular should it be?
+        date_format=None,
+        # Deprecated options kept for now for backwards compatibility
         include_month=False,
         include_day=False,
     ):
-        validate_time_period_options(**locals())
-        return "with_death_recorded_in_cpns", locals()
+        return "with_death_recorded_in_cpns", process_arguments(locals())
 
     @staticmethod
     def date_of(
-        source, include_month=False, include_day=False, return_expectations=None
+        source, return_expectations=None,
+        date_format=None,
+        # Deprecated options kept for now for backwards compatibility
+        include_month=False,
+        include_day=False,
     ):
         returning = "date"
-        return "value_from", locals()
+        return "value_from", process_arguments(locals())
+
+
+def process_arguments(args):
+    validate_time_period_options(**args)
+
+    for date_arg in ("reference_date", "start_date", "end_date"):
+        if date_arg in args:
+            value = args[date_arg]
+            if value == "today":
+                value = datetime.date.today()
+            args[date_arg] = datetime.date.fromisoformat(str(value))
+
+    # Handle deprecated API
```

Comment:
Is there any value in raising a deprecation warning here? I think on balance there probably is; though on the other hand we only have two studies at the moment and could just go and fix them and them remove the compatibility stuff

Reply:
I'd be more inclined to update the studies and remove the BC code. I included BC because it wasn't much extra work and it makes the whole process much easier, but I'd prefer not to carry it around long term.